### PR TITLE
Added original_case to the dns_request event

### PR DESCRIPTION
**Pull request by rvictory**
This change adds a new field to the dns_request event that contains the DNS query in the original letter case that the client requested it as. For example, if I query for "EXAMPLE.com" the "query" field will still be "example.com" but the original_query field will be "EXAMPLE.com"

I also added a new policy script that will add this field to the dns.log log file.

Btests were created that cover both changes and all Btests pass (minus the documentation Btest, which says that I can ignore the fact that it failed).

Hopefully this is decent for my first PR!

Reference issue: https://github.com/zeek/zeek/issues/1013

### CODE REVIEW

**Comment by rvictory** [comment]
I'm having a lot of trouble understanding why the checks are failing. It appears to be something related to a CI-based test that includes the DNS log, but I'm not sure why/how my new policy script is being loaded. Is there anybody that help my troubleshoot?

**Comment by jsiwek** [comment]
CI runs an additional test suite in `testing/external` that ends up loading that `test-all-policy.zeek` script (now changed in the PR to load `log-original-query-case.zeek `).

If you wanted to run and double-check those external tests yourself:

```
cd testing/external
make init
cd zeek-testing
make
```

You could get the PR to "pass" CI by changing `PUBLIC_REPO` in `testing/external/Makefile` to point to your fork w/ updated baselines, along with changing `testing/external/commit-hash.zeek-testing` to the associate commit hash, but don't think it's worth the trouble in this case.  The test failures here mostly just give maintainers a heads up of "hey, you should check and update the external test suite baselines when merging".

**Comment by rvictory** [comment]
Thanks @jsiwek! I'm running the external tests locally now. Should be easy enough to fix the baseline <and> the external tests very clearly show that the original_query casing is correct!

**Comment by rvictory** [comment]
It definitely appears that if the external Baselines were updated to include the original_query field, that would be enough to get the tests to pass.

File: src/analyzer/protocol/dns/DNS.cc [hunk 1]
```diff
@@ -196,9 +196,16 @@ bool DNS_Interpreter::ParseQuestion(DNS_MsgInfo* msg,
 
 	if ( dns_event && ! msg->skip_event )
 		{
-		BroString* question_name =
-			new BroString(name, name_end - name, true);
-		SendReplyOrRejectEvent(msg, dns_event, data, len, question_name);
+	    BroString* original_name = new BroString(name, name_end - name, true);
```
Comment (JustinAzoff) [inline src/analyzer/protocol/dns/DNS.cc:199]: The spaces here should be tabs

File: src/analyzer/protocol/dns/DNS.cc [hunk 2]
```diff
@@ -375,9 +388,10 @@ u_char* DNS_Interpreter::ExtractName(const u_char*& data, int& len,
 		}
 
 	// Convert labels to lower case for consistency.
-	for ( u_char* np = name_start; np < name; ++np )
-		if ( isupper(*np) )
-			*np = tolower(*np);
+	if (downcase)
+        for ( u_char* np = name_start; np < name; ++np )
```
Comment (JustinAzoff) [inline src/analyzer/protocol/dns/DNS.cc:392]: Another place where tabs got changed to spaces

File: src/analyzer/protocol/dns/DNS.cc (hunk 1 above)
Comment (rvictory) [reply src/analyzer/protocol/dns/DNS.cc]: https://github.com/zeek/zeek/pull/1015/commits/7a91b49f5a25ce91575383cd45cdd4bba971515a

File: src/analyzer/protocol/dns/DNS.cc (hunk 2 above)
Comment (rvictory) [reply src/analyzer/protocol/dns/DNS.cc]: https://github.com/zeek/zeek/pull/1015/commits/7a91b49f5a25ce91575383cd45cdd4bba971515a

**Comment by rvictory** [comment]
Tabs should be fixed in https://github.com/zeek/zeek/pull/1015/commits/7a91b49f5a25ce91575383cd45cdd4bba971515a

**Comment by rvictory** [comment]
All looks good, thanks @jsiwek !

**Pull request merged**